homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -175,6 +175,40 @@ def _setup_from_config(self, config):
         self._supported_features |= config[CONF_XY] and SUPPORT_COLOR
         self._supported_features |= config[CONF_HS] and SUPPORT_COLOR
 
+    def _parse_color(self, values):
+        try:
+            red = int(values["color"]["r"])
+            green = int(values["color"]["g"])
+            blue = int(values["color"]["b"])
+
+            return color_util.color_RGB_to_hs(red, green, blue)
+        except KeyError:
+            pass
+        except ValueError:
+            _LOGGER.warning("Invalid RGB color value received")
```

Comment:
We should stop execution when this happens. If you send us values, they should be valid.